File: gto-support-viewpager/src/main/java/org/ccci/gto/android/common/viewpager/util/ViewPagerUtils.java
package org.ccci.gto.android.common.viewpager.util;

import androidx.annotation.NonNull;

import org.ccci.gto.android.common.util.view.ViewUtils;

/**
 * @deprecated Since v1.2.2, use {@link org.ccci.gto.android.common.util.view.ViewUtils} instead.
 */
@Deprecated
public class ViewPagerUtils {
    /**
     * @deprecated Since v1.2.2, use
     * {@link org.ccci.gto.android.common.util.view.ViewUtils#handleOnInterceptTouchEventException(Throwable)} instead.
     */
    @Deprecated
    public static <T extends Throwable> boolean handleOnInterceptTouchEventException(@NonNull final T cause) throws T {
        return ViewUtils.handleOnInterceptTouchEventException(cause);
    }
}

File: gto-support-viewpager/src/main/java/org/ccci/gto/android/common/viewpager/util/ViewPagerUtils.kt
@file:JvmName("ViewPagerUtils")

package org.ccci.gto.android.common.viewpager.util

import android.widget.Scroller
import androidx.viewpager.widget.ViewPager
import timber.log.Timber

fun ViewPager.setScroller(scroller: Scroller) {
    try {
        ViewPager::class.java.getField("mScroller")
            .apply { isAccessible = true }
            .set(this, scroller)
    } catch (e: Exception) {
        Timber.tag("ViewPagerUtils")
            .e(e, "Error setting custom scroller on ViewPager")
    }
}
